h5pt.go

```diff
@@ -12,26 +12,37 @@ package hdf5
 import "C"
 
 import (
+	"encoding/binary"
 	"fmt"
+	"math"
 	"reflect"
 	"runtime"
 	"unsafe"
 )
 
+var nativeEndian binary.ByteOrder
+
+func init() {
+	val := uint16(0x1)
+	valPtr := (*[2]byte)(unsafe.Pointer(&val))
+
+	if valPtr[0] == 1 {
+		nativeEndian = binary.LittleEndian
+	} else {
+		nativeEndian = binary.BigEndian
+	}
+}
+
 // appendData is a wrapper type for information necessary to create and
 // subsequently write an in memory object to a PacketTable using Append().
 type appendData struct {
-	ptr        unsafe.Pointer
-	memSize    C.size_t
-	offset     C.size_t
-	numRecords C.size_t
+	cBuffer    []byte
```

Comment:
I think we could have a name that's more informative than `cBuffer`. It's a buffer of bytes, so it can just be called `buf`. The `c` actually detracts from the name unless it means something other than character, in which case it's not informative enough and should be spelled out.